lua/luasnip/extras/fmt.lua

```diff
@@ -0,0 +1,159 @@
+local ls = require('luasnip')
+
+-- Format a string using a syntax similar to Python ''.format() method.
+--
+-- Placeholders are enclosed in `{}` and can be:
+-- * with numbers, e.g. `{1}` - takes `args` elemengs by index, e.g. `args[1]`
+-- * with names, e.g. `{myname}` - takes `args` elements by key, e.g. `args['myname']`
+-- * empty, e.g. `{}`, like with numbers, but takes the next index, if no numbered
+--   placeholders have been used, then it starts indexing from 1, else the value
+--   of last numbered placeholder + 1 is used
+-- To input regular curly braces use `{{}}` or `{{text}}`. The text between braces
+-- is not matched by [0-9a-zA-Z_-]* then single braces can be used, no need to escape.
+--
+-- Arguments:
+--   fmt: string
+--   args: table
+-- Returns: a list of strings and elements of `args` inserted into placeholders.
+local function replace_placeholders(fmt, args)
+    local pattern = '{[0-9a-zA-Z_-]*}'
```

Comment:
Does this work for eg. `{{ } }}`? What do you think of separately escaping opening and closing brackets via `\` or `%`?

Reply:
Actually currently this would ignore everything in these examples, as the pattern does not contain spaces. I think that we don't really need spaces in placeholders and this reduces the number of escaping the user has to do in most scenarios (like all the `{ int x = 1; }` scopes in C-like languages.

Reply:
And maybe it's enough to use single braces, as in practice it should rarely happen that a snippet contains braces that are not a placeholder and contain a text without any spaces inside. What do you think?

Reply:
Ah right, bad example :sweat_smile:
Mhmm, I think we should in any case provide a way to properly escape `{` and `}`, (in eg. latex `\begin{something}` is commonplace).
In general, I'd prefer always having to escape brackets (if they should be in the text literally) to having some rules on when it's necessary and when it isn't, but that's just my preference, maybe being smart about it is nicer for most users.

Reply:
Now brackets must always be escaped